File: RI_Stack/apps/qa/hn/testapp/src/org/cablelabs/test/TestRunner.java
// COPYRIGHT_BEGIN
//  DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER
//  
//  Copyright (C) 2008-2013, Cable Television Laboratories, Inc. 
//  
//  This software is available under multiple licenses: 
//  
//  (1) BSD 2-clause 
//   Redistribution and use in source and binary forms, with or without modification, are
//   permitted provided that the following conditions are met:
//        ·Redistributions of source code must retain the above copyright notice, this list 
//             of conditions and the following disclaimer.
//        ·Redistributions in binary form must reproduce the above copyright notice, this list of conditions 
//             and the following disclaimer in the documentation and/or other materials provided with the 
//             distribution.
//   THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS 
//   "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED 
//   TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A 
//   PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT 
//   HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, 
//   SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT 
//   LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, 
//   DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY 
//   THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT 
//   (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF 
//   THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
//  
//  (2) GPL Version 2
//   This program is free software; you can redistribute it and/or modify
//   it under the terms of the GNU General Public License as published by
//   the Free Software Foundation, version 2. This program is distributed
//   in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
//   even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR
//   PURPOSE. See the GNU General Public License for more details.
//  
//   You should have received a copy of the GNU General Public License along
//   with this program.If not, see<http:www.gnu.org/licenses/>.
//  
//  (3)CableLabs License
//   If you or the company you represent has a separate agreement with CableLabs
//   concerning the use of this code, your rights and obligations with respect
//   to this code shall be as set forth therein. No license is granted hereunder
//   for any other purpose.
//  
//   Please contact CableLabs if you need additional information or 
//   have any questions.
//  
//       CableLabs
//       858 Coal Creek Cir
//       Louisville, CO 80027-9750
//       303 661-9100
// COPYRIGHT_END

package org.cablelabs.test;

import java.io.File;
import java.io.FileNotFoundException;
import java.util.Vector;

import javax.tv.util.TVTimer;
import javax.tv.util.TVTimerSpec;
import javax.tv.util.TVTimerWentOffEvent;
import javax.tv.util.TVTimerWentOffListener;

public class TestRunner implements TVTimerWentOffListener
{
    private static final String TEST_LIST_TOKEN = "TEST_CATAGORY:";

    private static final String TEST_DELIM = ",";

    private static final String TEST_COMMENT = "#";

    private Logger m_logger = null;

    private ElementList m_rootList = null;

    private int m_state = 0;

    private Test m_currentTest = null;

    private Thread m_testThread = null;

    private Vector m_stateChangeListeners = null;

    private Vector m_statusListeners = null;

    private TVTimer m_testTimer = null;

    private TVTimerSpec m_timerSpec = null;

    public TestRunner(String testFileName) throws FileNotFoundException
    {
        m_state = TestingState.TESTING_STATE_IDLE;
        m_currentTest = null;
        m_testThread = null;
        m_stateChangeListeners = new Vector();
        m_statusListeners = new Vector();

        File aFile = new File(testFileName);

        m_logger = new Logger();
        m_logger.setPrefix("TestRunner: ");
        m_logger.log("TestFile name = " + testFileName);

        TestElementBuilder builder = new TestElementBuilder();
        builder.buildTestListFromStream(aFile);
        m_rootList = builder.getRootElements();
    }

    public TestRunner()
    {

        m_state = TestingState.TESTING_STATE_IDLE;
        m_currentTest = null;
        m_testThread = null;
        m_stateChangeListeners = new Vector();
        m_statusListeners = new Vector();

        m_logger = new Logger();
        m_logger.setPrefix("TestRunner: ");
    }

    public void runTests(String[] testNames)
    {
        TestElementBuilder builder = new TestElementBuilder();
        ElementList list = null;// builder.buildTests(testNames)
        if (null != list)
        {
            this.runCategory(list);
        }
    }

    public void executeTest(final Test t)
    {
        m_currentTest = t;
        m_testTimer = TVTimer.getTimer();
        m_timerSpec = new TVTimerSpec();
        m_timerSpec.addTVTimerWentOffListener(this);
        m_timerSpec.setTime(System.currentTimeMillis() + t.getTimeout());
        try
        {
            m_testTimer.scheduleTimerSpec(m_timerSpec);
        }
        catch (Exception e)
        {
            e.printStackTrace();
        }
        m_testThread = new Thread()
        {

            public void run()
            {
                TestRunner.this.m_logger.log("executeTest.run() - test: " + t.getName());
                runTest(t);
                changeStatus(t.getStatus(), t);
            }
        };
        m_testThread.start();
    }

    public void timerWentOff(TVTimerWentOffEvent e)
    {
        if (m_testThread != null && m_testThread.isAlive())
        {
            // Hmmmmm - not sure how to kill this thread
            // m_testThread.stop();
            m_testThread = null;
            changeState(TestingState.TESTING_STATE_INTERRUPTED, m_currentTest);
            m_currentTest.setStatus(Test.TEST_STATUS_INTERRUPTED);
        }
    }

    public void dumpTests(ElementList list)
    {
        m_logger.log("dumpTests Enter - parentList = "
                + ((null != list.getParent()) ? list.getParent().getName() : "none"));
        Vector rootlist = list.getElements();
        for (int ii = 0; ii < rootlist.size(); ii++)
        {
            if (rootlist.elementAt(ii) instanceof Test)
            {
                Test t = (Test) rootlist.elementAt(ii);
                m_logger.log("Test: " + t.getName());
            }
            else
            {
                dumpTests((ElementList) rootlist.elementAt(ii));
            }
        }
        m_logger.log("dumpTests Exit.");
    }

    public static void main(String args[])
    {
        System.out.print("Main enter");

        try
        {
            TestRunner runner = new TestRunner("c:\\tests.txt");

            runner.dumpTests(runner.getRootElementList());
            ElementList catList = runner.getRootElementList();
            runner.runCategory(catList);
        }
        catch (FileNotFoundException fnfe)
        {
            fnfe.printStackTrace();
        }
        System.out.print("Main exit");
    }

    public void runCategory(ElementList list)
    {
        Vector testElements = list.getElements();
        for (int ii = 0; ii < testElements.size(); ii++)
        {
            if (testElements.elementAt(ii) instanceof ElementList)
            {
                m_logger.log("Found list: " + ((ElementList) testElements.elementAt(ii)).getName()
                        + " running Catagory...");
                runCategory((ElementList) testElements.elementAt(ii));
            }
            else
            {
                Test t = (Test) testElements.elementAt(ii);
                m_logger.log("Executing test: " + t.getName());
                runTest(t);
                changeStatus(t.getStatus(), t);
                m_logger.log("Test: " + t.getName());
                m_logger.log("    Description: " + t.getDescription());
                m_logger.log("    Status: " + Test.statusToString(t.getStatus()));
            }
        }

    }

    public void setCurrentTest(Test t) throws IllegalArgumentException
    {
        if (t == null)
        {
            throw new IllegalArgumentException("Invalid test: " + t);
        }
        else
        {
            m_currentTest = t;
            return;
        }
    }

    public void runTest(Test t)
    {

        int result = Test.TEST_STATUS_NONE;

        if (t == null)
        {
            changeState(TestingState.TESTING_STATE_IDLE, t);
            return;
        }

        this.m_logger.log("runTest Enter - test: " + t.getName());

        changeState(TestingState.TESTING_STATE_PREPARING, t);
        result = t.prepare();
        t.setStatus(result);
        // changeStatus(result, t);
        if (result != Test.TEST_STATUS_PASS)
        {
            changeState(TestingState.TESTING_STATE_IDLE, t);
        }

        changeState(TestingState.TESTING_STATE_EXECUTING, t);
        result = t.execute();
        t.setStatus(result);
        // changeStatus(result, t);
        if (result != Test.TEST_STATUS_PASS)
        {
            changeState(TestingState.TESTING_STATE_IDLE, t);
            return;
        }

        changeState(TestingState.TESTING_STATE_CLEANING, t);
        result = t.clean();
        t.setStatus(result);
        // changeStatus(result, t);

        changeState(TestingState.TESTING_STATE_IDLE, t);
        this.m_testTimer.deschedule(this.m_timerSpec);
    }

    public int getCurrentTestStatus()
    {
        return (this.m_currentTest == null) ? Test.TEST_STATUS_NONE : this.m_currentTest.getStatus();
    }

    public void addTestStateChangeListener(TestStateChangeListener listener)
    {
        m_stateChangeListeners.add(listener);
    }

    public void removeTestStateChangeListener(TestStateChangeListener listener)
    {
        m_stateChangeListeners.remove(listener);
    }

    public void addTestStatusListener(TestStatusListener listener)
    {
        this.m_statusListeners.add(listener);
    }

    public void removeTestStatusListener(TestStatusListener listener)
    {
        m_statusListeners.remove(listener);
    }

    private void changeState(int newState, Test t)
    {
        for (int ii = 0; ii < m_stateChangeListeners.size(); ii++)
        {
            TestStateChangeListener listener = (TestStateChangeListener) m_stateChangeListeners.elementAt(ii);
            listener.notifyStateChanged(newState, m_state, t);
        }

        m_state = newState;
    }

    private void changeStatus(int status, Test t)
    {
        t.setStatus(status);
        for (int ii = 0; ii < m_statusListeners.size(); ii++)
        {
            TestStatusListener listener = (TestStatusListener) m_statusListeners.elementAt(ii);
            listener.notifyStatusChanged(status, t);
        }
    }

    public ElementList getRootElementList()
    {
        return this.m_rootList;
    }

    public String getCurrentTestName()
    {
        return (this.m_currentTest == null) ? "NONE" : this.m_currentTest.m_name;
    }

}

File: RI_Stack/mpe/test/os/generic/TestRunner.c
// COPYRIGHT_BEGIN
//  DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER
//  
//  Copyright (C) 2008-2013, Cable Television Laboratories, Inc. 
//  
//  This software is available under multiple licenses: 
//  
//  (1) BSD 2-clause 
//   Redistribution and use in source and binary forms, with or without modification, are
//   permitted provided that the following conditions are met:
//        ·Redistributions of source code must retain the above copyright notice, this list 
//             of conditions and the following disclaimer.
//        ·Redistributions in binary form must reproduce the above copyright notice, this list of conditions 
//             and the following disclaimer in the documentation and/or other materials provided with the 
//             distribution.
//   THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS 
//   "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED 
//   TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A 
//   PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT 
//   HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, 
//   SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT 
//   LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, 
//   DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY 
//   THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT 
//   (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF 
//   THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
//  
//  (2) GPL Version 2
//   This program is free software; you can redistribute it and/or modify
//   it under the terms of the GNU General Public License as published by
//   the Free Software Foundation, version 2. This program is distributed
//   in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
//   even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR
//   PURPOSE. See the GNU General Public License for more details.
//  
//   You should have received a copy of the GNU General Public License along
//   with this program.If not, see<http:www.gnu.org/licenses/>.
//  
//  (3)CableLabs License
//   If you or the company you represent has a separate agreement with CableLabs
//   concerning the use of this code, your rights and obligations with respect
//   to this code shall be as set forth therein. No license is granted hereunder
//   for any other purpose.
//  
//   Please contact CableLabs if you need additional information or 
//   have any questions.
//  
//       CableLabs
//       858 Coal Creek Cir
//       Louisville, CO 80027-9750
//       303 661-9100
// COPYRIGHT_END

#include <stdarg.h>
#include <mpetest_dbg.h>
#include <test_sys.h>
#include <test_net.h>
#include <mpe_error.h>
#include <mpeos_file.h>

/*
 #include <mpeos_dbg.h>
 #include <dbgmgr.h>
 */

void **mpe_ftable; /* Dummy reference to resolve bogus dependency. */

/*  Functions defined in this source file  */

void TestRunner(void);
mpe_Bool vte_agent_Log(const char* format, ...);

/* External functions used in this source file  */

extern void mpeos_envInit(void);
extern void mpeos_dbgInit(void);
extern void mpeos_dbgLogControlInit(void);

extern void test_sysRunFpTests(void);

/**
 * TestRunner
 *
 * This is the entry point from main.c.  This function provides a
 * simple compile-time selection of test cases to execute.
 *
 */
void TestRunner(void)
{
    const char *errmsg = NULL;

    vte_agent_Log(
            "\n\n##########\n##########\n#\n#   Into 'TestRunner()'\n#\n#\n");

    /*
     *  Init logging system.
     *
     *  WARNING : Don't try to use logging system before it's init'ed!
     */

    mpeos_envInit();
    mpeos_dbgInit();
    mpeos_dbgLogControlInit();

    /*
     *  The following controls logging when running tests. Adjust the "TEST"
     *  logging level to get the desired amount of test log output. INFO
     *  and FATAL should always be enabled for testing. Also enable DEBUG and
     *  TRACE1 if you want to see all possible output.
     */

    /* Start by turning off all TEST logging, so we start from a known place  */

    errmsg = mpeos_dbgLogControlOpSysIntf("TEST", "NONE");
    if (errmsg == NULL)
    {
        vte_agent_Log("'mpeos_dbgLogControlOpSysIntf()' returned NULL");
    }
    else if (!strcmp(errmsg, " OK"))
    {
        vte_agent_Log("'mpeos_dbgLogControlOpSysIntf()' returned '%s'\n",
                errmsg);
    }

    /*
     *  Turn on desired test logging. For normal test runs use "INFO FATAL"
     *  When doing test development use "INFO FATAL DEBUG TRACE1" or
     *  "ALL DEBUG TRACE" to see all test output
     */

    errmsg = mpeos_dbgLogControlOpSysIntf("TEST", "INFO FATAL TRACE1 TRACE5");
    //    errmsg = mpeos_dbgLogControlOpSysIntf("TEST", "FATAL DEBUG INFO TRACE1 TRACE2 TRACE3 TRACE4 TRACE5 TRACE6 TRACE7 TRACE8 TRACE9");
    if (errmsg == NULL)
    {
        vte_agent_Log("'mpeos_dbgLogControlOpSysIntf()' returned NULL");
    }
    else if (!strcmp(errmsg, " OK"))
    {
        vte_agent_Log("'mpeos_dbgLogControlOpSysIntf()' returned '%s'\n",
                errmsg);
    }
    else
    {
        MPEOS_LOG(MPE_LOG_INFO, MPE_MOD_TEST,
                "\n'TestRunner()' - Logging enabled :\n");
    }

    /*
     *  For debugging you can enable logging here for other modules, for example :
     *
     *     errmsg = mpeos_dbgLogControlOpSysIntf("MEDIA", "FATAL ERROR WARN INFO DEBUG");
     */

    /*

     mpeos_dbgLogControlOpSysIntf("GFX", "ALL DEBUG TRACE");

     mpeos_dbgLogControlOpSysIntf("DIRECTFB", "ALL DEBUG TRACE");

     mpeos_dbgLogControlOpSysIntf("DISP", "ALL DEBUG TRACE");

     */
    /*
     *  The following print the test logging levels. Comment out if you
     *  don't want to see the messages.
     *
     */

    MPEOS_LOG(MPE_LOG_FATAL, MPE_MOD_TEST,
            "  MPE_MOD_TEST - MPE_LOG_FATAL enabled\n");
    MPEOS_LOG(MPE_LOG_ERROR, MPE_MOD_TEST,
            "  MPE_MOD_TEST - MPE_LOG_ERROR enabled\n");
    MPEOS_LOG(MPE_LOG_WARN, MPE_MOD_TEST,
            "  MPE_MOD_TEST - MPE_LOG_WARN enabled\n");
    MPEOS_LOG(MPE_LOG_INFO, MPE_MOD_TEST,
            "  MPE_MOD_TEST - MPE_LOG_INFO enabled\n");
    MPEOS_LOG(MPE_LOG_DEBUG, MPE_MOD_TEST,
            "  MPE_MOD_TEST - MPE_LOG_DEBUG enabled\n");
    MPEOS_LOG(MPE_LOG_TRACE1, MPE_MOD_TEST,
            "  MPE_MOD_TEST - MPE_LOG_TRACE1 enabled\n");
    MPEOS_LOG(MPE_LOG_TRACE2, MPE_MOD_TEST,
            "  MPE_MOD_TEST - MPE_LOG_TRACE2 enabled\n");
    MPEOS_LOG(MPE_LOG_TRACE3, MPE_MOD_TEST,
            "  MPE_MOD_TEST - MPE_LOG_TRACE3 enabled\n");
    MPEOS_LOG(MPE_LOG_TRACE4, MPE_MOD_TEST,
            "  MPE_MOD_TEST - MPE_LOG_TRACE4 enabled\n");
    MPEOS_LOG(MPE_LOG_TRACE5, MPE_MOD_TEST,
            "  MPE_MOD_TEST - MPE_LOG_TRACE5 enabled\n");
    MPEOS_LOG(MPE_LOG_TRACE6, MPE_MOD_TEST,
            "  MPE_MOD_TEST - MPE_LOG_TRACE6 enabled\n");
    MPEOS_LOG(MPE_LOG_TRACE7, MPE_MOD_TEST,
            "  MPE_MOD_TEST - MPE_LOG_TRACE7 enabled\n");
    MPEOS_LOG(MPE_LOG_TRACE8, MPE_MOD_TEST,
            "  MPE_MOD_TEST - MPE_LOG_TRACE8 enabled\n");
    MPEOS_LOG(MPE_LOG_TRACE9, MPE_MOD_TEST,
            "  MPE_MOD_TEST - MPE_LOG_TRACE9 enabled\n");

    /*
     *  If you enable logging for other modules and want to know if it's
     *  really working, you can add similar calls for the other modules here :
     *
     *
     *    MPEOS_LOG (MPE_LOG_FATAL,  MPE_MOD_MEDIA, "MPE_MOD_MEDIA - MPE_LOG_FATAL enabled\n");
     *    MPEOS_LOG (MPE_LOG_ERROR,  MPE_MOD_MEDIA, "MPE_MOD_MEDIA - MPE_LOG_ERROR enabled\n");
     *    MPEOS_LOG (MPE_LOG_WARN,   MPE_MOD_MEDIA, "MPE_MOD_MEDIA - MPE_LOG_WARN enabled\n");
     *    MPEOS_LOG (MPE_LOG_INFO,   MPE_MOD_MEDIA, "MPE_MOD_MEDIA - MPE_LOG_INFO enabled\n");
     *    MPEOS_LOG (MPE_LOG_DEBUG,  MPE_MOD_MEDIA, "MPE_MOD_MEDIA - MPE_LOG_DEBUG enabled\n");
     */

    MPEOS_LOG(MPE_LOG_INFO, MPE_MOD_TEST,
            "\n  'TestRunner()' - launching tests\n\n");

    test_netRunAllTests();
    test_sysRunAllTests(); /* execute all of the system tests  */

    //    test_sysRunFpTests();

    /*
     * To execute just an individual group of the system tests use
     * one of the following functions:
     *

     test_sysRunTimeTests();
     test_sysRunEventTests();
     test_sysRunThreadTests();
     test_sysRunMemTests();
     test_sysRunSyncTests();
     test_sysRunUtilTests();

     */

    /*
     * To execute an individual test within one of the system test 
     * groups use one of the following functions:
     *

     test_sysRunTimeGetTest();
     test_sysRunTimeGetMillisTest();
     test_sysRunTimeToDateTest();
     test_sysRunTimeClockTest();
     test_sysRunTimeClockTicksTest();
     test_sysRunTimeClockToMillisTest();
     test_sysRunTimeClockToTimeTest();
     test_sysRunTimeMillisToClockTest();
     test_sysRunTimeSystemClockTest();
     test_sysRunTimeTimeToClockTest();
     test_sysRunTimeTmToTimeTest();

     test_sysRunEventPingPongTest();
     test_sysRunEventQueueDeleteTest();
     test_sysRunEventQueueNewTest();
     test_sysRunEventQueueNextTest();
     test_sysRunEventQueueSendTest();
     test_sysRunEventQueueWaitNextTest();
     test_sysRunEventSendThreadTest();

     test_sysRunThreadAttachTest();
     test_sysRunThreadCreateTest();
     test_sysRunThreadDestroyID0Test();
     test_sysRunThreadDestroyIDTest();
     test_sysRunThreadDestroyOtherTest();
     test_sysRunThreadGetCurrentTest();
     test_sysRunThreadGetDataTest();
     test_sysRunThreadGetStatusTest();
     test_sysRunThreadSetDataTest();
     test_sysRunThreadSetPriorityTest();
     test_sysRunThreadSetStatusTest();
     test_sysRunThreadSleepTest();
     test_sysRunThreadYieldTest();
     test_sysRunThreadDataTest();
     test_sysRunThreadStatusTest();

     test_sysRunMemAllocPTest();
     test_sysRunMemFreePTest();
     test_sysRunMemReallocPTest();
     test_sysRunMemAllocHTest();
     test_sysRunMemFreeHTest();
     test_sysRunMemLockHTest();
     test_sysRunMemReallocHTest();
     test_sysRunMemCompactTest();
     test_sysRunMemGetFreeSizeTest();
     test_sysRunMemGetLargestFreeTest();
     test_sysRunMemGetStatsTest();
     test_sysRunMemPurgeTest();
     test_sysRunMemStatsTest();

     test_sysRunSyncMutexNewTest();
     test_sysRunSyncMutexAcquireTest();
     test_sysRunSyncMutexAcquierTryTest();
     test_sysRunSyncCondNewTest();
     test_sysRunSyncCondGetSetTest();
     test_sysRunSyncCondPingPongTest();
     test_sysRunSyncCondWaitForTest();

     test_sysRunUtilEnvGet();
     test_sysRunUtilEnvInit();
     test_sysRunUtilEnvSet();
     test_sysRunUtilIniSetPath();
     test_sysRunUtilLongJmp();
     test_sysRunUtilRegisterForPowerKey();
     test_sysRunUtilSetJmp();
     test_sysRunUtilStbBoot();
     test_sysRunUtilStbBootStatus();
     test_sysRunUtilStbGetAcOutletState();
     test_sysRunUtilStbGetPowerStatus();
     test_sysRunUtilStbGetRootCerts();
     test_sysRunUtilStbSetAcOutletStateTest();

     */

    /*
     * To execute just an individual group of the network tests use
     * one of the following functions:
     *

     test_netRunLookupTests();
     test_netRunConnectTests();
     test_netRunOptionsTests();
     test_netRunSelectTests();
     test_mpe_netReadWriteTests();

     */

    MPEOS_LOG(
            MPE_LOG_INFO,
            MPE_MOD_TEST,
            "\n\n#\n#\n#   'TestRunner()' is finished.\n#\n################\n################\n\n");

    return;
}

/**
 * vte_agent_Log - Print raw mwssages on console. Used before logging system
 * is inited. Adjust as needed to do raw text output on STB console.
 *
 *   NOTE : Since this does things differently for different STBs, it probably
 *          should live somewhere besides in the generic 'TestRunner'.
 *
 */

mpe_Bool vte_agent_Log(const char* format, ...)
{
    va_list arg;
    va_start(arg, format);
    vprintf(format, arg);
    va_end(arg);
    return TRUE;
}

/****
 *
 *  Fake ROMFS stuff, to allow linking of MPEOS tests
 *
 */

static void fake_mpe_fileRomfsNamev(const char*);
static mpe_FileError fake_mpe_fileRomfsName(const char*);
static mpe_FileError fake_mpe_fileRomfsOpen(const char*, mpe_FileOpenMode,
        mpe_File*);
static mpe_FileError fake_mpe_fileRomfs_handle(mpe_File);
static mpe_FileError fake_mpe_fileRomfs_RW(mpe_File, uint32_t*, void*);
static mpe_FileError fake_mpe_fileRomfs_stat(const char*, mpe_FileStatMode,
        mpe_FileInfo *);
static mpe_FileError fake_mpe_fileRomfs_fstat(mpe_File, mpe_FileStatMode,
        mpe_FileInfo*);
static mpe_FileError fake_mpe_fileRomfs_seek(mpe_File, mpe_FileSeekMode,
        int64_t*);
static mpe_FileError fake_mpe_fileRomfsReName(const char*, const char*);
static mpe_FileError fake_mpe_fileRomfs_dopen(const char*, mpe_Dir*);
static mpe_FileError fake_mpe_fileRomfs_dread(mpe_Dir, mpe_DirEntry*);
static mpe_FileError fake_mpe_fileRomfs_dclose(mpe_Dir);
static mpe_FileError fake_mpe_fileRomfs_dirMount(const mpe_DirUrl*);
static mpe_FileError fake_mpe_fileRomfs_dirUStat(const mpe_DirUrl*,
        mpe_DirStatMode, mpe_DirInfo*);

mpeos_filesys_ftable_t mpe_fileRomfsFTable =
{ fake_mpe_fileRomfsNamev, // mpe_fileRomfsInit, 
        fake_mpe_fileRomfsOpen, // mpe_fileRomfsFileOpen, 
        fake_mpe_fileRomfs_handle, // mpe_fileRomfsFileClose, 
        fake_mpe_fileRomfs_RW, // mpe_fileRomfsFileRead, 
        fake_mpe_fileRomfs_RW, // mpe_fileRomfsFileWrite, 
        fake_mpe_fileRomfs_seek, // mpe_fileRomfsFileSeek, 
        fake_mpe_fileRomfs_handle, // mpe_fileRomfsFileSync, 
        fake_mpe_fileRomfs_stat, // mpe_fileRomfsFileGetStat, 
        fake_mpe_fileRomfs_stat, // mpe_fileRomfsFileSetStat, 
        fake_mpe_fileRomfs_fstat, // mpe_fileRomfsFileGetFStat, 
        fake_mpe_fileRomfs_fstat, // mpe_fileRomfsFileSetFStat, 
        fake_mpe_fileRomfsName, // mpe_fileRomfsFileDelete, 
        fake_mpe_fileRomfsReName, // mpe_fileRomfsFileRename, 
        fake_mpe_fileRomfs_dopen, // mpe_fileRomfsDirOpen, 
        fake_mpe_fileRomfs_dread, // mpe_fileRomfsDirRead, 
        fake_mpe_fileRomfs_dclose, // mpe_fileRomfsDirClose, 
        fake_mpe_fileRomfsName, // mpe_fileRomfsDirDelete, 
        fake_mpe_fileRomfsReName, // mpe_fileRomfsDirRename, 
        fake_mpe_fileRomfsName, // mpe_fileRomfsDirCreate, 
        fake_mpe_fileRomfs_dirMount, // mpe_fileRomfsDirMount, 
        fake_mpe_fileRomfs_dirMount, // mpe_fileRomfsDirUnmount, 
        fake_mpe_fileRomfs_dirUStat, // mpe_fileRomfsDirGetUStat, 
        fake_mpe_fileRomfs_dirUStat, // mpe_fileRomfsDirSetUStat 
        };

static void fake_mpe_fileRomfsNamev(const char* name)
{
    MPEOS_LOG(MPE_LOG_INFO, MPE_MOD_TEST, "\nfileRomfsName() - '%s'\n", name);
}

static mpe_FileError fake_mpe_fileRomfsName(const char* name)
{
    MPEOS_LOG(MPE_LOG_TRACE3, MPE_MOD_TEST, "\nfileRomfsName() - '%s'\n", name);
}

static mpe_FileError fake_mpe_fileRomfsOpen(const char* name,
        mpe_FileOpenMode openMode, mpe_File* returnHandle)
{
    MPEOS_LOG(MPE_LOG_INFO, MPE_MOD_TEST, "\nfileRomfsOpen() - '%s'\n", name);
}

static mpe_FileError fake_mpe_fileRomfs_handle(mpe_File handle)
{
    MPEOS_LOG(MPE_LOG_INFO, MPE_MOD_TEST, "\nfake_mpe_fileRomfs_handle()\n");
}

static mpe_FileError fake_mpe_fileRomfs_RW(mpe_File handle, uint32_t* count,
        void* buffer)
{
    MPEOS_LOG(MPE_LOG_INFO, MPE_MOD_TEST, "\nfake_mpe_fileRomfs_RW()\n");
}

static mpe_FileError fake_mpe_fileRomfs_seek(mpe_File handle,
        mpe_FileSeekMode seekMode, int64_t* offset)
{
    MPEOS_LOG(MPE_LOG_INFO, MPE_MOD_TEST, "\nfake_mpe_fileRomfs_seek()\n");
}

static mpe_FileError fake_mpe_fileRomfs_stat(const char* fileName,
        mpe_FileStatMode mode, mpe_FileInfo *info)
{
    MPEOS_LOG(MPE_LOG_INFO, MPE_MOD_TEST, "\nfake_mpe_fileRomfs_stat()\n");
}

static mpe_FileError fake_mpe_fileRomfs_fstat(mpe_File handle,
        mpe_FileStatMode mode, mpe_FileInfo *info)
{
    MPEOS_LOG(MPE_LOG_INFO, MPE_MOD_TEST, "\nfake_mpe_fileRomfs_fstat()\n");
}

static mpe_FileError fake_mpe_fileRomfsReName(const char* oldName,
        const char* newName)
{
    MPEOS_LOG(MPE_LOG_INFO, MPE_MOD_TEST, "\nfileRomfsReName() - '%s'\n",
            oldName);
}

static mpe_FileError fake_mpe_fileRomfs_dopen(const char* name,
        mpe_Dir* returnHandle)
{
    MPEOS_LOG(MPE_LOG_INFO, MPE_MOD_TEST, "\nfileRomfsDirOpen() - '%s'\n", name);
}

static mpe_FileError fake_mpe_fileRomfs_dread(mpe_Dir handle,
        mpe_DirEntry* dirEnt)
{
    MPEOS_LOG(MPE_LOG_INFO, MPE_MOD_TEST, "\nfileRomfsDirRead()\n");
}

static mpe_FileError fake_mpe_fileRomfs_dclose(mpe_Dir handle)
{
    MPEOS_LOG(MPE_LOG_INFO, MPE_MOD_TEST, "\nfileRomfsDirClose()\n");
}

static mpe_FileError fake_mpe_fileRomfs_dirMount(const mpe_DirUrl *dirUrl)
{
    MPEOS_LOG(MPE_LOG_INFO, MPE_MOD_TEST, "\nfileRomfsDirMount()\n");
}

static mpe_FileError fake_mpe_fileRomfs_dirUStat(const mpe_DirUrl *dirUrl,
        mpe_DirStatMode mode, mpe_DirInfo *info)
{
    MPEOS_LOG(MPE_LOG_INFO, MPE_MOD_TEST, "\nfileRomfsDirUStat()\n");
}

